sycl/include/CL/sycl/intel/bit_cast.hpp

```diff
@@ -0,0 +1,44 @@
+//==----------- bit_cast.hpp --- SYCL bit_cast -----------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/defines.hpp>
+#include <CL/sycl/detail/helpers.hpp>
+
+#include <numeric>
+#include <type_traits>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+
+template <typename To, typename From>
+constexpr To bit_cast(const From &from) noexcept {
```

Comment:
Interesting that extension is named `INTEL`, while the function placed into `sycl` namespace: I wonder whether we need to rename the extension spec or not.

Reply:
Done.